drivers/hmis/app/jobs/hmis/migrate_assessments_job.rb

```diff
@@ -0,0 +1,154 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class MigrateAssessmentsJob < ApplicationJob
+    include Hmis::Concerns::HmisArelHelper
+
+    # Construct CustomAssessment and FormProcessor records for Assessment-related records.
+    # Records considered to be sourced from the same assessment if they share an Information date
+    # and Data Collection Stage.
+    #
+    # This mostly ASSUMES that the data quality is perfect. If data is bad, it will created bad assessments.
+    # For example:
+    # * Exit Date should match the Information Date for other records collected at Exit.
+    # * There should be only 1 group of records collected at Entry, all with the same Information date
+    # * There should be only 1 group of records collected at Exit, all with the same Information date
```

Comment:
@eanders do you think these are reasonable expectations? This could definitely be made more robust. My concerns is creating multiple Entry or Exit assessments if there are multiple records with different information dates.

Reply:
Is there any reason not to use DataCollectionStage to determine which are attached to Entry / Exit?  This is probably fine, but it would be safer to use DataCollectionStage and then group non-at entry/at exit by information date.

Reply:
It does use DataCollectionStage. Sorry that isn't very clear in the comment "at Entry" means DCS is 1.

I think there IS a reason not to use _Information Date_, though, because it could result in multiple entry or exit assmts being created. Say for example we have a set of records with DCS 1 that have varying information dates, they really should end up on the same Intake assessment, but they are being split out as different assessments because they have different information dates. That would be "bad data" but I'm not sure if we should expect it.

Reply:
Maybe that's exactly what you were saying. Entry and Exit records should be grouped by DCS _only_, and other stages should be grouped by DCS+Information Date. It would be safer

Reply:
Yep, that's what I was trying to say.  Entry and Exit records should be treated specially and we should enforce only one of each.  The others should be grouped by DCS and information date.

Reply:
OK, I updated as such.